target/linux/ramips/image/mt76x8.mk

```diff
@@ -71,12 +71,21 @@ define Device/glinet_vixmini
 endef
 TARGET_DEVICES += glinet_vixmini
 
-define Device/hc5661a
+define Device/hiwifi_hc5661a
   DTS := HC5661A
-  IMAGE_SIZE := $(ralink_default_fw_size_16M)
+  IMAGE_SIZE := 15808k
+  SUPPORTED_DEVICES := hc5661a
```

Comment:
In theory this will break sysupgrade on hc5661a. (I don't have time to verify my assumption though.)
sysupgrade checks board name against SUPPORTED_DEVICES and since you've changed board name to hiwifi,hc5661a this match will fail.
It should be SUPPORTED_DEVICES += hc5661a instead.